runtimes/pythonrt/runner_manager_local.go

```diff
@@ -0,0 +1,226 @@
+package pythonrt
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"sync"
+	"time"
+
+	"go.uber.org/zap"
+
+	pb "go.autokitteh.dev/autokitteh/proto/gen/go/autokitteh/remote/v1"
+)
+
+type localRunnerManager struct {
+	logger           *zap.Logger
+	pyExe            string
+	runnerIDToRunner map[string]*LocalPython
+	mu               *sync.Mutex
+	workerAddress    string
+}
+
+type LocalRunnerConfig struct {
+	WorkerAddress string
+}
+
+func ConfigureLocalRunnerManager(log *zap.Logger, cfg LocalRunnerConfig) error {
+	// if configuredRunnerType != runnerTypeNotConfigured {
+	// 	if configuredRunnerType == runnerTypeLocal {
+	// 		return nil
+	// 	}
+	// 	return errors.New("runner type already configured, cannot configure twice")
+	// }
+
+	pyExe, isUserPy, err := pythonToRun(log)
+	if err != nil {
+		return err
+	}
+
+	const timeout = 3 * time.Second
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	info, err := pyExeInfo(ctx, pyExe)
+	if err != nil {
+		return err
+	}
+
+	log.Info("python info", zap.String("exe", info.Exe), zap.Any("version", info.Version))
+	if !isGoodVersion(info.Version) {
+		const format = "python >= %d.%d required, found %q"
+		return fmt.Errorf(format, minPyVersion.Major, minPyVersion.Minor, info.VersionString)
+	}
+	lm := &localRunnerManager{
+		logger:           log,
+		runnerIDToRunner: map[string]*LocalPython{},
+		mu:               new(sync.Mutex),
+		workerAddress:    cfg.WorkerAddress,
+	}
+
+	// If user supplies which Python to use, we use it "as-is" without creating venv
+	if !isUserPy {
+		if err := ensureVEnv(log, pyExe); err != nil {
+			return fmt.Errorf("create venv: %w", err)
+		}
+		lm.pyExe = venvPy
+	} else {
+		lm.pyExe = pyExe
+	}
+	log.Info("using python", zap.String("exe", lm.pyExe))
+
+	configuredRunnerType = runnerTypeLocal
+	runnerManager = lm
+	return nil
+}
+
+func (l *localRunnerManager) Start(ctx context.Context, buildArtifacts []byte, vars map[string]string) (string, pb.RunnerClient, error) {
+	r := &LocalPython{
+		log: l.logger,
+	}
+
+	if err := r.Start(l.pyExe, buildArtifacts, vars, l.workerAddress); err != nil {
+		return "", nil, err
+	}
+
+	runnerAddr := fmt.Sprintf("localhost:%d", r.port)
+	client, err := dialRunner(runnerAddr)
+	if err != nil {
+		if err := r.Close(); err != nil {
+			l.logger.Warn("close runner", zap.Error(err))
+		}
+		return "", nil, err
+	}
+
+	l.mu.Lock()
+	l.runnerIDToRunner[r.id.String()] = r
+	l.mu.Unlock()
+	return r.id.String(), client, nil
+}
+
+func (*localRunnerManager) RunnerHealth(ctx context.Context, runnerID string) error { return nil }
```

Comment:
Check that PID is up (you can do `kill 0` to the process to see if it's alive)

Reply:
done  + added check loop